apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemDAO.java

```diff
@@ -112,25 +116,29 @@ INSERT INTO dataset_items (
                     ) AS created_by,
                     new.last_updated_by
                 FROM (
-                    SELECT
-                        :id AS id,
-                        :datasetId AS dataset_id,
-                        :source AS source,
-                        :traceId AS trace_id,
-                        :spanId AS span_id,
-                        :input AS input,
-                        :expectedOutput AS expected_output,
-                        :metadata AS metadata,
-                        now64(9) AS created_at,
-                        :workspace_id AS workspace_id,
-                        :createdBy AS created_by,
-                        :lastUpdatedBy AS last_updated_by
+                    <items:{item |
+                         SELECT
+                             :id<item.index> AS id,
+                             :datasetId<item.index> AS dataset_id,
+                             :source<item.index> AS source,
+                             :traceId<item.index> AS trace_id,
+                             :spanId<item.index> AS span_id,
+                             :input<item.index> AS input,
+                             :expectedOutput<item.index> AS expected_output,
+                             :metadata<item.index> AS metadata,
+                             now64(9) AS created_at,
+                             :workspace_id<item.index> AS workspace_id,
+                             :createdBy<item.index> AS created_by,
+                             :lastUpdatedBy<item.index> AS last_updated_by
+                         <if(item.hasNext)>
+                            UNION ALL
+                         <endif>
+                    }>
                 ) AS new
                 LEFT JOIN (
                     SELECT
                         *
                     FROM dataset_items
-                    WHERE id = :id
```

Comment:
This is going to be problematic as it will result in a full scan of the table.

Reply:
That is good point I will add workspace at least